### DISCUSSION TITLE
Availability of apt/apt-get or cifs-utils in Linux images?

### DISCUSSION THREAD

**Discussion by mrlife** [dotnet/dotnet-docker#5521] [Q&A]
Hi, are apt, apt-get, or cifs-utils available on any of the [SDK images](https://github.com/dotnet/dotnet-docker/blob/main/README.sdk.md#full-tag-listing)? I have tried a few, including 8.0-noble, but don't see them. According to [this page](https://github.com/dotnet/dotnet-docker/blob/main/documentation/image-variants.md#composite-net-80), the ".NET images for Ubuntu ... are full-featured". With the "-noble" variant being Ubuntu, I'm wondering if I'm missing something. Thank you!

**Comment by mthalman** [comment]
`apt`/`apt-get` is definitely available.

```
$ docker run --rm mcr.microsoft.com/dotnet/sdk:8.0-noble apt-get --version
apt 2.7.14 (amd64)
Supported modules:
*Ver: Standard .deb
 Pkg:  Debian APT solver interface (Priority -1000)
 Pkg:  Debian APT planner interface (Priority -1000)
*Pkg:  Debian dpkg interface (Priority 30)
 S.L: 'deb' Debian binary tree
 S.L: 'deb-src' Debian source tree
 Idx: EDSP scenario file
 Idx: EIPP scenario file
 Idx: Debian Source Index
 Idx: Debian Package Index
 Idx: Debian Translation Index
 Idx: Debian dpkg status file
 Idx: Debian deb file
 Idx: Debian dsc file
 Idx: Debian control file
```

`cifs-utils` is not installed by default because it's not installed in the base Ubuntu image and .NET has no dependency on it. You can install it yourself with `apt-get`.

**Reply by mrlife** [reply]
Thanks, @mthalman. When I try to install `cifs-utils`, it can't find the package. All I can think is that it doesn't have access to the internet...does a docker container need configured for internet access? The docker is running in a CI/CD pipeline.

> $ apt-get install cifs-utils
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package cifs-utils

**Reply by mthalman** [reply]
You likely haven't run `apt-get update` first. That's required so the tool knows what packages are available from the package repositories. You may find this guide useful: https://itsfoss.com/apt-get-linux-guide.

**Reply by mrlife** [reply]
@mthalman, you are absolutely right. Thank you, that allowed the process to continue.  I don't know if I can ask a followup question here, but I wanted to try, as I've been stuck on something else.

After installing `cifs-util`, I am attempting to mount a Windows share, however the connection is being blocked with error: 

> CIFS VFS: Error connecting to socket. Aborting operation.
CIFS VFS: cifs_mount failed w/return code = -115

I'm sure that the container is not known to the domain Windows server is on, so that server is blocking it. Do you know this is usually resolved?

**Reply by mthalman** [reply]
Sorry, I'm not knowledgeable on CIFS. This might be a better question for https://unix.stackexchange.com or https://serverfault.com as it is unrelated to .NET.
